packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -852,23 +861,35 @@ class _ToggleButton extends StatelessWidget {
   /// The button's label, which is usually an [Icon] or a [Text] widget.
   final Widget child;
 
+  Color _resolveColor(Set<MaterialState> states, MaterialStateProperty<Color?> widgetColor,
```

Comment:
Please use MaterialStateProperty.resolveAs as suggested in https://github.com/flutter/flutter/pull/82026#discussion_r650181183.  You could improve it a little, since _DefaultFillColor is known the be a `MaterialStateProperty<Color>`: 

```dart
Color _resolveColor(Set<MaterialState> states, Color? widgetColor, Color? themeColor, MaterialStateProperty<Color> defaultColor) {
  return MaterialStateProperty.resolveAs<Color?>(widgetColor, states)
    ?? MaterialStateProperty.resolveAs<Color?>(themeColor, states)
    ?? defaultColor.resolve(states);
}
```

Reply:
I'm afraid, we can't use `resolveAs` directly, As described in [MaterialStateProperty.resolveAs](https://github.com/flutter/flutter/blob/d79295af24c3ed621c33713ecda14ad196fd9c31/packages/flutter/lib/src/material/material_state.dart#L533), If `widgetColor` is a `color` property it will return the actual value itself, irrespective of material states.

The [test](https://github.com/flutter/flutter/blob/4392a46fc11b0818db1341914f9b3640c27a7555/packages/flutter/test/material/toggle_buttons_test.dart#L668) will fail due to an unselected button using the selected button's color which supposed to use the default color.

Even more slimplier test that fails 
```dart
testWidgets('Test will fail', (WidgetTester tester) async {
    const Color customFillColor = Colors.green;
    await tester.pumpWidget(
      Material(
        child: boilerplate(
          child: ToggleButtons(
            fillColor: customFillColor,
            isSelected: const <bool>[false],
            onPressed: (int index) {},
            children: <Widget>[
              Row(children: const <Widget>[
                Text('First child'),
              ]),
            ],
          ),
        ),
      ),
    );

    final Material material = tester.widget<Material>(find.descendant(
      of: find.byType(RawMaterialButton),
      matching: find.byType(Material),
    ));
    expect(material.color, isNot(customFillColor));
    expect(material.type, MaterialType.button);
  });
```

Reply:
Good point; the existing theme and widget fill color properties are for "selected toggle buttons".  _ResolveFillColor looks good then.